apis/v1alpha1/generator.yaml

```diff
@@ -51,6 +51,64 @@ resources:
         - ValidationError
         - ValidationException
         - InvalidEventPatternException
+  Endpoint:
+    fields:
+      Name:
+        is_immutable: true
+        is_required: true
+      Description:
+        compare:
+          is_ignored: true
+      RoleARN:
+        compare:
+          is_ignored: true
+      EventBuses:
+        compare:
+          is_ignored: true
+      ReplicationConfig:
+        compare:
+          is_ignored: true
+      StateReason:
+        is_read_only: true
+        from:
+          operation: DescribeEndpoint
+          path: StateReason
+    tags:
+      ignore: true
+    hooks:
+      delta_pre_compare:
+        code: customPreCompare(delta, a, b)
+      sdk_create_pre_build_request:
+        template_path: hooks/endpoint/sdk_create_pre_build_request.go.tpl
+      sdk_create_post_set_output:
+        template_path: hooks/endpoint/sdk_create_post_set_output.go.tpl
+      sdk_update_pre_build_request:
+        template_path: hooks/endpoint/sdk_update_pre_build_request.go.tpl
+      sdk_update_post_build_request:
+        template_path: hooks/endpoint/sdk_update_post_build_request.go.tpl
+      sdk_update_pre_set_output:
+        template_path: hooks/endpoint/sdk_update_pre_set_output.go.tpl
+      sdk_delete_pre_build_request:
+        template_path: hooks/endpoint/sdk_delete_pre_build_request.go.tpl
+      sdk_delete_post_request:
+        template_path: hooks/endpoint/sdk_delete_post_request.go.tpl
+    print:
+      add_age_column: true
+      add_synced_column: true
+      additional_columns:
+        - name: ARN
+          json_path: .status.ackResourceMetadata.arn
+          type: string
```

Comment:
IMO ARNs are a little bit too verbose for a simple `kubectl  get`, maybe we can print only when a user uses `-o wide` ? https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/config/field.go#L236-L239

Reply:
Done in new commit